src/mlpack/methods/kmeans/hamerly_kmeans_impl.hpp

```diff
@@ -118,20 +129,24 @@ double HamerlyKMeans<DistanceType, MatType>::Iterate(const arma::mat& centroids,
         lowerBounds(i) = dist;
       }
     }
-    distanceCalculations += centroids.n_cols - 1;
+    threadDistanceCalculations += centroids.n_cols - 1;
 
     // Update new centroids.
     newCentroids.col(assignments[i]) += dataset.col(i);
     ++counts(assignments[i]);
   }
 
+  distanceCalculations += threadDistanceCalculations;
```

Comment:
If we're reducing `threadDistanceCalculations`, then couldn't we just use `distanceCalculations` directly?